backend/src/plugins/offboard/offboard_service_impl.h

```diff
@@ -0,0 +1,116 @@
+#include "plugins/offboard/offboard.h"
+#include "offboard/offboard.grpc.pb.h"
+
+namespace dronecode_sdk {
+namespace backend {
+
+template<typename Offboard = Offboard>
+class OffboardServiceImpl final : public rpc::offboard::OffboardService::Service {
+public:
+    OffboardServiceImpl(Offboard &offboard) : _offboard(offboard) {}
+
+    template<typename ResponseType>
+    void fillResponseWithResult(ResponseType *response,
+                                dronecode_sdk::Offboard::Result &offboard_result) const
+    {
+        auto rpc_result = static_cast<rpc::offboard::OffboardResult::Result>(offboard_result);
+
+        auto *rpc_offboard_result = new rpc::offboard::OffboardResult();
+        rpc_offboard_result->set_result(rpc_result);
+        rpc_offboard_result->set_result_str(dronecode_sdk::Offboard::result_str(offboard_result));
+
+        response->set_allocated_offboard_result(rpc_offboard_result);
+    }
+
+    grpc::Status Start(grpc::ServerContext * /* context */,
+                       const rpc::Offboard::StartRequest * /* request */,
+                       rpc::Offboard::StartResponse *response) override
+    {
+        auto offboard_result = _offboard.start();
+
+        if (response != nullptr) {
+            fillResponseWithResult(response, offboard_result);
+        }
+
+        return grpc::Status::OK;
+    }
+
+    grpc::Status Stop(grpc::ServerContext * /* context */,
+                      const rpc::Offboard::StopRequest * /* request */,
+                      rpc::Offboard::StopResponse *response) override
+    {
+        auto offboard_result = _offboard.stop();
+
+        if (response != nullptr) {
+            fillResponseWithResult(response, offboard_result);
+        }
+
+        return grpc::Status::OK;
+    }
+
+    grpc::Status IsActive(grpc::ServerContext * /* context */,
+                          const rpc::Offboard::IsActiveRequest * /* request */,
+                          rpc::Offboard::IsActiveResponse *response) override
+    {
+        if (response != nullptr) {
+            auto is_active = _offboard.is_active();
+            response->set_is_active(is_active);
+        }
+
+        return grpc::Status::OK;
+    }
+
+    grpc::Status SetAttitudeRate(grpc::ServerContext * /* context */,
+                                 const rpc::Offboard::SetAttitudeRateRequest * /* request */,
```

Comment:
And then you need to follow the compiler warnings here. E.g. `request` is required in this method.